lightningd/jsonrpc.h

```diff
@@ -101,5 +102,9 @@ json_tok_address_scriptpubkey(const tal_t *ctx,
 			      const char *buffer,
 			      const jsmntok_t *tok, const u8 **scriptpubkey);
 
+/* A specialized get_params used for both withdraw and fundchannel */
+bool json_wtx_get_params(struct wallet_tx * tx, const char * buffer,
```

Comment:
I would suggest using a `json_tok_wtx` rather than replacing `json_get_params`: we may want to add different arguments to `fundchannel` vs `withdraw` in the future.  So you would retain `json_get_params` but replace the satoshi-parsing with `json_tok_wtx` instead.

Reply:
Very good. I'm on it.